ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/util/PDEJavaHelper.java

```diff
@@ -286,9 +287,16 @@ private static IPackageFragment searchWorkspaceForPackage(String packageName, IP
 					continue;
 				}
 				IJavaProject jp = JavaCore.create(projects[i]);
+				Map<IPath, IPackageFragmentRoot> rootsByPath = new HashMap<>();
+				for (IPackageFragmentRoot classpathRoot : jp.getAllPackageFragmentRoots()) {
+					IPath classRootPath = classpathRoot.getPath();
+					if (classRootPath != null) {
+						rootsByPath.put(classRootPath, classpathRoot);
+					}
+				}
 				ListIterator<IPath> li = libPaths.listIterator();
 				while (li.hasNext()) {
```

Comment:
Since you already touched this, could you please convert this while loop and if you want the outer for loop over all projects into an enhanced for loop?
Btw. I wonder why there is no quick-fix that suggest to convert this iterator+while into an enhanced for-loop.

Reply:
there is a cleanup, that could do this. but it currently fails. let's wait till that is fixed: https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/798